pkg/operator/hive/manageddns.go

```diff
@@ -37,38 +42,105 @@ func (r *ReconcileHiveConfig) teardownLegacyExternalDNS(hLog log.FieldLogger) er
 	return nil
 }
 
-func deployManagedDomainsConfigMap(h *resource.Helper, instance *hivev1.HiveConfig) error {
-	cm := &corev1.ConfigMap{}
-	cm.Kind = "ConfigMap"
-	cm.APIVersion = "v1"
-	cm.Name = managedDomainsConfigMapName
-	cm.Namespace = constants.HiveNamespace
+func (r *ReconcileHiveConfig) teardownLegacyMangedDomainsConfigMap(hLog log.FieldLogger) error {
+	key := client.ObjectKey{Namespace: constants.HiveNamespace, Name: "managed-domains"}
+	if err := resource.DeleteAnyExistingObject(r, key, &corev1.ConfigMap{}, hLog); err != nil {
+		hLog.WithError(err).Error("failed to delete legacy managed domains configmap")
+	}
+	return nil
+}
+
+// configureManagedDomains will create a new configmap holding the managed domains settings (if necessary), or simply
+// return the current configmap of the current deployment if the settings it contains match the desired settings.
+// It will also return the old configmap if a new one was created (so that it can be garbage collected).
+func (r *ReconcileHiveConfig) configureManagedDomains(logger log.FieldLogger, instance *hivev1.HiveConfig) (*corev1.ConfigMap, error) {
+	var mdConfigMap *corev1.ConfigMap
 
 	domains, err := json.Marshal(instance.Spec.ManagedDomains)
 	if err != nil {
-		return errors.Wrap(err, "failed to marshal managed domains list into the configmap")
+		return nil, errors.Wrap(err, "failed to marshal managed domains list into the configmap")
 	}
 
-	cm.Data = map[string]string{"managed-domains": string(domains)}
-	_, err = h.ApplyRuntimeObject(cm, scheme.Scheme)
-	return errors.Wrap(err, "error applying managed domains configmap")
+	newConfigMapData := map[string]string{managedDomainsConfigMapKey: string(domains)}
+
+	currentConfigMap, err := r.getCurrentConfigMap(newConfigMapData, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	if !reflect.DeepEqual(newConfigMapData, currentConfigMap.Data) {
```

Comment:
We have already done this deep equal check in `getCurrentConfigMap`. There is no need to do it again. If you return nil from `getCurrentConfigMap` when there is no good configmap, then you can check for nil here to determine whether you need to create a new configmap.